src/sentry/api/serializers/models/organization_member/expand/teams.py

```diff
@@ -13,19 +13,22 @@ def get_attrs(
     ) -> MutableMapping[OrganizationMember, MutableMapping[str, Any]]:
         attrs = super().get_attrs(item_list, user)
 
-        team_ids_by_organization_member_id = get_team_slugs_by_organization_member_id(item_list)
+        teams, teams_with_role = get_teams_by_organization_member_id(item_list)
         for item in item_list:
-            teams = team_ids_by_organization_member_id.get(item.id, [])
+
             try:
-                attrs[item]["teams"] = teams
+                attrs[item]["teams"] = teams.get(item.id, [])  # Deprecated
+                attrs[item]["teamRoles"] = teams_with_role.get(item.id, [])
             except KeyError:
-                attrs[item] = {"teams": teams}
+                attrs[item] = {"teams": teams}  # Deprecated
```

Comment:
@leedongwei it looks like this line of code is useless since `attrs[item]` will be overwritten with a new dictionary on the next line. 🤔